scripts/apps/authoring/attachments/actions.ts

```diff
@@ -0,0 +1,89 @@
+import {gettext} from 'core/utils';
+
+export const INIT_ATTACHMENTS = 'INIT_ATTACHMENTS';
+export function initAttachments(item) {
+    return (dispatch, geState, {deployConfig, attachments}) => {
+        const payload = {
+            maxSize: deployConfig.getSync('attachments_max_size'),
+            maxFiles: deployConfig.getSync('attachments_max_files'),
+            files: [],
+        };
+
+        attachments.byItem(item)
+            .then((files) => {
+                payload.files = files;
```

Comment:
How about initializing the state object and dispatching right here. Mutating objects or using `finally` is not needed then.

```
attachments.byItem(item)
    .then((files) => {
        dispatch({type: INIT_ATTACHMENTS, payload: {
            maxSize: deployConfig.getSync('attachments_max_size'),
            maxFiles: deployConfig.getSync('attachments_max_files'),
            files: files,
        }});
    });
```

Reply:
yep that makes sense, when there is an error it should probably do something else instead of showing empty list